### PR TITLE
Bectu

### PR DESCRIPTION
**Pull request by jackcarlisle**
# ready for review

- [x] adds bectu tv fields 
- [x] splits offer creation into tabs

### CODE REVIEW

File: lib/ev2/production/calculation_helpers.ex [hunk 1]
```diff
@@ -116,6 +117,241 @@ defmodule Ev2.Production.CalculationHelpers do
...
+    ((calculate_network_days(start_date, end_date, working_week) / 5) / 1)
+  end
+
+  def calculate_total(list) do
+    Enum.sum(list)
+  end
+
+  def calculate_overtime_inc_hol(daily_rate) do
+    IO.inspect daily_rate
+    IO.inspect ((daily_rate / 10) * 1.5)
```
Comment (JMurphyWeb) [inline lib/ev2/production/calculation_helpers.ex]: two inspects

File: lib/ev2/production/offer/offer.ex [hunk 2]
```diff
@@ -223,13 +363,71 @@ defmodule Ev2.Production.Offer do
...
+      offer_params,
+      :equipment_rental_required?,
+      @equipment_rental_fields
+    )
+    |> validate_length(:equipment_rental_description, max: 300)
+    |> validate_length(:box_rental_description, max: 300)
+  end
+
+  def update_allowances_changeset(struct, offer_params, project) do
+    offer = Production.get_offer(struct.id)
```
Comment (JMurphyWeb) [inline lib/ev2/production/offer/offer.ex]: isn't the offer already the struct?

Comment (jackcarlisle) [reply lib/ev2/production/offer/offer.ex]: Sorry yes will change that back

File: lib/ev2_web/controllers/offer_controller.ex [hunk 3]
```diff
@@ -219,25 +233,40 @@ defmodule Ev2Web.OfferController do
...
+    case section do
+      "details" -> proj_offer_path(conn, :edit_tv, project, offer)
+      "tv" -> proj_offer_path(conn, :edit_allowances, project, offer)
+      "allowances" ->
+        case length(project_custom_offer_fields) == 0 do
+          true ->
+            proj_offer_path(conn, :show, project, offer)
+            false ->
+              proj_offer_custom_field_path(conn, :add, project, offer)
+            end
```
Comment (JMurphyWeb) [inline lib/ev2_web/controllers/offer_controller.ex]: weird indenting here

File: lib/ev2/production/offer/offer_api.ex [hunk 4]
```diff
@@ -317,22 +328,42 @@ defmodule Ev2.Production.OfferAPI do
...
       """
-      def offer_form_valid?(%{"offer" => offer_params, "proj_id" => project_id}) do
-        changeset = validate_offer(%Offer{}, offer_params)
+      def offer_form_valid?(offer, %{"offer" => offer_params, "proj_id" => project_id}, section) do
+        struct =
+          case offer do
+            nil -> %Offer{}
+            offer -> offer
+          end
+        project = Repo.get(Project, project_id)
```
Comment (JMurphyWeb) [inline lib/ev2/production/offer/offer_api.ex]: Do you think we can pass project in as an argument to `offer_form_valid?`? I think we already have it whenever this function is called so this line would be an unnecessary second call to the db.

File: lib/ev2_web/router.ex [hunk 5]
```diff
@@ -107,8 +107,10 @@ defmodule Ev2Web.Router do
       resources "/offers", OfferController, param: "offer_id", except: [:delete]
       post "/offers/:offer_id/send_offer", OfferController, :send_offer
       # scoped offer paths
-      get "/offers/:offer_id/download-file", OfferController, :download_supporting_document
-
+      get "/offers/:offer_id/offers/:offer_id/download-file", OfferController, :download_supporting_document
```
Comment (JMurphyWeb) [inline lib/ev2_web/router.ex]: don't know what's happened here

Comment (jackcarlisle) [reply lib/ev2_web/router.ex]: think this was in the merge conflicts. I'll add that one back

File: lib/ev2/production/offer/offer_api.ex (hunk 4 above)
Comment (jackcarlisle) [reply lib/ev2/production/offer/offer_api.ex]: yeah sounds good

File: lib/ev2/production/offer/offer.ex [hunk 6]
```diff
@@ -230,13 +370,70 @@ defmodule Ev2.Production.Offer do
...
+    job = offer.job_title
+    job_title = job.name
+    department = job.department.name
+    offer_params =
+      offer_params
+      |> Map.put_new("job", job_title)
+      |> Map.put_new("department", department)
+      |> Map.put_new("daily_or_weekly", offer.daily_or_weekly)
+    {contract, job} = get_contract_and_job(offer_params, project)
+    offer
```
Comment (JMurphyWeb) [inline lib/ev2/production/offer/offer.ex]: I think you can call the above function here, and remove the next 15 or so lines?

Comment (jackcarlisle) [reply lib/ev2/production/offer/offer.ex]: yeah good call

**Pull request merged**